### PR TITLE
57/refactor canvas molecule code

### PR DESCRIPTION
**Pull request by hagoptaminian**
a few notes:

- Let me know if you have any comments on the guided tour/its language
- Might be worth specifically reviewing the `Tour.js` component, `appInstanceResources.js`, and `StudentMode.js` to double check that the `appInstanceResource` dispatch is correct (though I tested it extensively)
- For #37 : I removed the 'Fill All' shortcut which wasn't visible and followed Philippe's suggestion that after selecting a side menu molecule and clicking on a canvas area, not to de-select the molecule (unless the canvas is full). This indeed makes it faster to fill the canvas.
- Cleaned up the Canvas molecule components for, hopefully, easier readability

### CODE REVIEW

File: src/components/lab/side-menu/molecules/SideMenuMethane.js [hunk 1]
```diff
@@ -33,17 +33,15 @@ const SideMenuMethane = () => {
...
-            marginLeft: '-2px',
+            marginBottom: '5px',
           }}
           charge={POSITIVE_CHARGE}
-          atomColor={HYDROGEN.atomColor.STANDARD}
+          atomColor={HYDROGEN.atomColor.DARKER}
         />
         <SideMenuHydrogen
-          customPositioningStyles={{ marginLeft: '-2px' }}
+          // customPositioningStyles={{ marginLeft: '-3px' }}
```
Comment (juancarlosfarah) [inline src/components/lab/side-menu/molecules/SideMenuMethane.js]: Remove this?

File: src/components/lab/canvas/molecules/CanvasWater.js [hunk 2]
```diff
@@ -4,96 +4,59 @@ import { Group } from 'react-konva';
...
-  const bottomHydrogenAtomCenterX = shouldOscillate
-    ? bottomHydrogenAtomInitialCenterX +
-      oscillationFactor * BOTTOM_HYDROGEN_AMPLITUDE
-    : bottomHydrogenAtomInitialCenterX;
-  const bottomHydrogenAtomCenterY =
-    y +
-    oxygenAtomRadius +
-    CANVAS_MOLECULES_DISTANCE_BETWEEN_VERTICAL_ATOMS +
-    hydrogenAtomRadius;
+  const topHydrogen = determineCoordinates(
```
Comment (juancarlosfarah) [inline src/components/lab/canvas/molecules/CanvasWater.js:22]: Mega improvement!

File: src/components/lab/canvas/CanvasMoleculeArea.js [hunk 3]
```diff
@@ -85,15 +85,28 @@ const CanvasMoleculeArea = ({
...
-      dispatch(selectMoleculeInSideMenu(null));
+      // when a molecule is selected in the side menu and added to the canvas, it remains selected
+      // this allows for quickly filling the canvas
+      // however, once the canvas is full, the side menu molecule should be de-selected
+      // this happens when (1) there were 3 molecules on the canvas (i.e. ...length - 1) and a 4th was added,
+      // (2) there were already 4 molecules on the canvas and a molecule was replaced
+      const numberOfMoleculesOnCanvas = moleculesOnCanvas.filter(
+        ({ molecule }) => molecule,
+      ).length;
+      if (
```
Comment (juancarlosfarah) [inline src/components/lab/canvas/CanvasMoleculeArea.js]: If you can't have more than `moleculesOnCanvas.length` molecules, you could simplify this with `numberOfMoleculesOnCanvas >= moleculesOnCanvas.length - 1`

File: src/components/lab/NonGreenhouseGases.js [hunk 4]
```diff
@@ -21,20 +20,9 @@ const NonGreenhouseGases = () => {
...
   return (
-    <GasesContainer
-      gasContainerLabel={t('Non-greenhouse Gases')}
-      showFillAllButton={
-        selectedMoleculeInSideMenu !== null &&
-        !GREENHOUSE_GASES.includes(selectedMoleculeInSideMenu) &&
-        canvasIncomplete
-      }
-    >
+    <GasesContainer gasContainerLabel={t('Non-greenhouse Gases')}>
```
Comment (juancarlosfarah) [inline src/components/lab/NonGreenhouseGases.js]: Not sure how it's in the `json`, but I would capitalise G, i.e. `Non-Greenhouse Gases`.

File: src/components/common/Tour.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+            marginRight: 10,
+            color: '#5050d2',
+            fontSize: '0.9em',
+          },
+          buttonNext: {
+            backgroundColor: '#5050d2',
+            fontSize: '0.9em',
+          },
+        }}
+        locale={{
```
Comment (juancarlosfarah) [inline src/components/common/Tour.js:110]: Not necessary now, but we should try to make this dynamic for more than 2 languages. And we should try to put all of the language strings in the same area (json files).

Comment (juancarlosfarah) [reply src/components/common/Tour.js:110]: You can probably achieve this using the `useTranslation` hook.

File: src/components/common/Tour.js [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+  const { standalone, lang } = useSelector(({ context }) => context);
+  const { content: appInstanceResourcesContent } = useSelector(
+    ({ appInstanceResources }) => appInstanceResources,
+  );
+  const INITIAL_TOUR_STATE = {
+    run: false,
+    continuous: true,
+    loading: false,
+    stepIndex: 0,
+    steps: lang === FRENCH_LANGUAGE_STRING ? TOUR_STEPS_FR : TOUR_STEPS_EN,
```
Comment (juancarlosfarah) [inline src/components/common/Tour.js:22]: See below about making dynamic for whatever language. In this case, you could probably use the `i18n` package where you define the steps.

Comment (juancarlosfarah) [reply src/components/common/Tour.js:22]: @hasanagh can provide an example, as I believe we did this for desktop.

**Review by juancarlosfarah — CHANGES_REQUESTED** [review]
Looking good! Much cleaner. Just some minor comments. Only tricky one is regarding language, which we should do, but can leave for another PR if necessary.

File: src/components/lab/side-menu/molecules/SideMenuMethane.js (hunk 1 above)
Comment (hagoptaminian) [reply src/components/lab/side-menu/molecules/SideMenuMethane.js]: oops

File: src/components/lab/canvas/CanvasMoleculeArea.js (hunk 3 above)
Comment (hagoptaminian) [reply src/components/lab/canvas/CanvasMoleculeArea.js]: 100%

File: src/components/common/Tour.js (hunk 6 above)
Comment (hagoptaminian) [reply src/components/common/Tour.js:22]: will be addressed in #59

File: src/components/common/Tour.js (hunk 5 above)
Comment (hagoptaminian) [reply src/components/common/Tour.js:110]: will be addressed in #59

**Pull request merged**